navit/main.c

```diff
@@ -405,7 +415,12 @@ void main_init(const char *program) {
     }
     if (!getenv("HOME"))
         setenv("HOME", getenv("NAVIT_PREFIX"), 0);
+#ifdef HAVE_API_WIN32_BASE && !HAVE_API_WIN32_CE
+    main_setup_environment(4);
+#endif
+#ifdef HAVE_API_WIN32_CE && !HAVE_API_WIN32_BASE
```

Comment:
What if for any reason (change of compiler defines in the future), we have either both `HAVE_API_WIN32_BASE` and `HAVE_API_WIN32_CE` or maybe none of them defined... we would then have no call at all to `main_setup_environment()`.
We are in the pre-condition _WIN32 || _WIN32_WCE, so in theory, we cannot assume anything about HAVE_API_WIN32_BASE or HAVE_API_WIN32_CE
Maybe we should just handle that case there, by adding a safeguard check like
```
#ifdef HAVE_API_WIN32_BASE && HAVE_API_WIN32_CE
#error Both HAVE_API_WIN32_BASE and HAVE_API_WIN32_CE cannot be defined simultaneously
#endif
#ifdef !HAVE_API_WIN32_BASE && !HAVE_API_WIN32_CE
#error Either HAVE_API_WIN32_BASE and HAVE_API_WIN32_CE should be defined here
#endif
```